### PR TITLE
plugin-tts: enhance TTS generation flow and caching

### PR DESCRIPTION
**Pull request by bfontes**
# Risks

**Risk Level:** Low  
**Details:** The changes primarily enhance the existing functionality of TTS generation. Potential risks include unexpected behavior in edge cases for input sanitization or caching logic.

# Background

## What does this PR do?

This PR introduces several improvements and optimizations to the TTS generation logic. Key changes include:

- Enhanced logging with proper levels (e.g., `debug` for queue updates and additional log messages for clarity).
- Improved input sanitization by refining prompt cleaning (e.g., removing specific commands and mentions).
- Integrated language detection to dynamically select a suitable voice for the prompt.
- Added caching logic to reuse previously generated audio files when available.
- Enhanced error handling with clear, user-friendly messages in failure scenarios.
- Updated validation to ensure the presence of required API keys.

## What kind of change is this?

- [x] Improvements (miscellaneous changes to existing features)

# Documentation changes needed?

- [X] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
  - **Details:** The TTS plugin documentation may need updates to reflect enhanced input sanitization, language detection, and caching behavior.

# Testing

## Where should a reviewer start?

1. Review the `generateTTS` function for updated logging, error handling, and TTS request flow.
2. Inspect changes to the `TTSGeneration` action, particularly input validation, prompt cleaning, and caching logic.
3. Verify updates in the `VOICE_MAP` logic for voice selection based on detected language.

## Detailed testing steps

1. As a user:
   - Provide a text prompt for TTS generation.
   - Verify the output includes an appropriate audio file or a clear error message.
2. Test edge cases:
   - Invalid prompts (e.g., fewer than three characters).
   - Caching reuse by submitting identical prompts.
   - Invalid or missing API keys.
3. Confirm proper logging at different levels (`debug`, `log`, `error`) in the console.

## Summary by CodeRabbit

- **Improvements**
	- Refined audio file caching mechanism for TTS generation
	- Enhanced logging practices with more precise debug-level logging
	- Optimized audio file handling process to improve efficiency

### CODE REVIEW

File: packages/plugin-tts/src/index.ts [hunk 1]
```diff
@@ -79,120 +77,87 @@
...
-            text: `I'll generate a audio based on your prompt: "${TTSPrompt}". This might take a few seconds...`,
+            text: `I'll generate an audio based on your prompt: "${TTSPrompt}". This might take a few seconds...`,
         });
 
         const language = detect(TTSPrompt);
         const voice_subject = VOICE_MAP[language[0].lang];
         const target_voice = getRandomVoice(voice_subject).fullName;
 
-        elizaLogger.log("Starting TTS generation with prompt:", prompt, "and voice:", target_voice);
+        elizaLogger.debug("Starting TTS generation with prompt:", TTSPrompt, "and voice:", target_voice);
```
[the remark this answers is not in the document]
Comment (bfontes) [reply packages/plugin-tts/src/index.ts]: Resolved at this commit `3097c6f275e834bb058fb48190dd8f0222cb9535`, could you check again, please?

**Review by wtfsayo — APPROVED** [review]
